src/app/drive/services/folder.service.ts

```diff
@@ -298,11 +298,18 @@ export async function downloadFolderAsZip({
   updateNumItems: () => void;
   options: DownloadFolderAsZipOptions;
   abortController?: AbortController;
-}): Promise<void> {
+}): Promise<{
+  totalItems: DriveFileData[] & SharedFiles[];
+  failedItems: DriveFileData[] & SharedFiles[];
+  allItemsFailed: boolean;
```

Comment:
Maybe adding this to types can help you to reuse it for the file, as I see that `DriveFileData[] & SharedFiles[]` are repeated.